### PR TITLE
Fixing PTDF balance constraint

### PR DESCRIPTION
**Pull request by sourabhdalvi**

### CODE REVIEW

File: src/network_models/ptdf_model.jl [hunk 1]
```diff
@@ -8,8 +8,7 @@ function ptdf_networkflow(
     time_steps = model_time_steps(psi_container)
     network_flow =
         add_cons_container!(psi_container, :network_flow, PTDF.axes[1], time_steps)
-    nodal_balance =
-        add_cons_container!(psi_container, :nodal_balance, PTDF.axes[2], time_steps)
+    nodal_balance = add_cons_container!(psi_container, :nodal_balance, time_steps)
```
Comment (jd-lara) [inline src/network_models/ptdf_model.jl:11]:
```suggestion
    nodal_balance = add_cons_container!(psi_container, :system_balance, time_steps)
```

File: src/network_models/ptdf_model.jl [hunk 2]
```diff
@@ -62,13 +61,10 @@ function ptdf_networkflow(
...
         end
 
-        for b in buses
-            number = PSY.get_number(b)
-            nodal_balance[number, t] = JuMP.@constraint(
-                psi_container.JuMPmodel,
-                nodal_balance_expressions[number, t] == 0
-            )
-        end
+        nodal_balance[t] = JuMP.@constraint(
```
Comment (jd-lara) [inline src/network_models/ptdf_model.jl:64]:
```suggestion
        system_balance[t] = JuMP.@constraint(
```

File: test/test_network_constructors.jl [hunk 3]
```diff
@@ -103,6 +104,8 @@ end
         psi_constraint_test(ps_model, constraint_names)
         psi_checkobjfun_test(ps_model, objfuncs[ix])
         psi_checksolve_test(ps_model, [MOI.ALMOST_OPTIMAL, MOI.OPTIMAL])
+        # TODO: add values from Bθ formulation to compare
```
Comment (jd-lara) [inline test/test_network_constructors.jl:107]: Please add this test before merging

File: test/test_network_constructors.jl [hunk 4]
```diff
@@ -162,6 +165,46 @@ end
     end
 end
 
+@testset "Test Locational Marginal Prices between DC lossless with PowerModels vs StandardPTDFModel" begin
+    network = [DCPPowerModel, StandardPTDFModel]
+    sys = build_system("c_sys5")
+    dual_constraint = [
+        [PSI.make_constraint_name(PSI.NODAL_BALANCE_ACTIVE, PSY.Bus)],
```
Comment (jd-lara) [inline test/test_network_constructors.jl:172]: It's better not to expose internal methods here. We should add a better interface to request the duals than passing the symbols but that is another issue.

File: test/test_network_constructors.jl [hunk 5]
```diff
@@ -162,6 +165,46 @@ end
...
+        ps_model = OperationsProblem(
+            TestOpProblem,
+            net,
+            sys;
+            optimizer = OSQP_optimizer,
+            use_parameters = p,
+            PTDF = ptdf,
+            constraint_duals = dual_constraint[ix],
+        )
+        construct_device!(ps_model, :Thermal, thermal_model)
```
Comment (jd-lara) [inline test/test_network_constructors.jl:189]: Why is this test calling `construct_device` here? Please use a template in the test.

**Review by jd-lara — CHANGES_REQUESTED** [review]
The code in the tests needs some refactoring to make it cleaner.

File: test/test_network_constructors.jl (hunk 5 above)
Comment (sourabhdalvi) [reply test/test_network_constructors.jl:189]: this is similar to what you have in other tests in that file, so I just followed that

File: test/test_network_constructors.jl (hunk 4 above)
Comment (sourabhdalvi) [reply test/test_network_constructors.jl:172]: I can replace it with just symbols

File: test/test_network_constructors.jl (hunk 3 above)
Comment (sourabhdalvi) [reply test/test_network_constructors.jl:107]: I added that test but just forgot to remove that TODO

File: test/test_network_constructors.jl (hunk 5 above)
Comment (jd-lara) [reply test/test_network_constructors.jl:189]: Some of the tests are old before the implementation of Templates, we need to refactor the testing to remove the individual construct device pattern

File: test/test_network_constructors.jl (hunk 4 above)
Comment (jd-lara) [reply test/test_network_constructors.jl:172]: Yes, that's better. This PR reflected some pending work with the interfaces

**Review by jd-lara — APPROVED** [review]
@sourabhdalvi can you change name of the constraint here please to make them consistent? Also, use the same syntax here calling ``add_cons_container!(psi_container, :system_balance, time_steps)`` instead of assign? 

https://github.com/NREL-SIIP/PowerSimulations.jl/blob/2b8689d3ea98722aefa5f7810a492488271a5fd6/src/network_models/copperplate_model.jl#L6

**Review by jd-lara — APPROVED** [review]

**Pull request merged**